File: connectx-backend/src/main/java/com/connectx/user/dto/UserIdentityKeyDto.java
package com.connectx.user.dto;

public class UserIdentityKeyDto {

    private String masterPublicKey;
    private String masterPrivateKey;

    public UserIdentityKeyDto() {}

    public UserIdentityKeyDto(String masterPublicKey, String masterPrivateKey) {
        this.masterPublicKey = masterPublicKey;
        this.masterPrivateKey = masterPrivateKey;
    }

    public String getMasterPublicKey() {
        return masterPublicKey;
    }

    public void setMasterPublicKey(String masterPublicKey) {
        this.masterPublicKey = masterPublicKey;
    }

    public String getMasterPrivateKey() {
        return masterPrivateKey;
    }

    public void setMasterPrivateKey(String masterPrivateKey) {
        this.masterPrivateKey = masterPrivateKey;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/user/model/UserIdentityKeyDto.kt
package com.connectx.app.data.remote.user.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.user.dto.UserIdentityKeyDto exactly. Both fields are opaque
 * strings to the server (see docs/CONNECTX_ANDROID_DEVELOPMENT.md Section 27/24 for
 * the E2EE private-key-custody discussion) -- this is a contract-only model, no
 * crypto handling exists anywhere in the Android project.
 */
@Serializable
data class UserIdentityKeyDto(
    val masterPublicKey: String? = null,
    val masterPrivateKey: String? = null
)
